### PR TITLE
fix: better loading for NewsFeed and FeedPostViewScreen

### PR DESCRIPTION
**Pull request by WaDadidou**

### CODE REVIEW

**Comment by omniwired** [comment]
@WaDadidou is it MUCH better than before.
Do you know if there is any way to limit these requests? cc @sakul-budhathoki 

Kapture 2023-04-11 at 11 49 40

**Review by omniwired — COMMENTED** [review]
@WaDadidou is it MUCH better than before.
Do you know if there is any way to limit these requests? cc @sakul-budhathoki 

Kapture 2023-04-11 at 11 49 40

File: packages/components/socialFeed/NewsFeed/NewsFeed.tsx [hunk 1]
```diff
@@ -38,8 +35,10 @@ export const NewsFeed: React.FC<NewsFeedProps> = ({
 }) => {
   const { data, isFetching, refetch, hasNextPage, fetchNextPage, isLoading } =
     useFetchFeed(req);
-  const isLoadingValue = useSharedValue(false);
-  const isGoingUp = useSharedValue(false);
+  const isLoadingValue = useMemo(
```
Comment (hthieu1110) [inline packages/components/socialFeed/NewsFeed/NewsFeed.tsx:38]: We can assign directly,  I think useMemo does not help much in this case and make the code longer :D

Comment (hthieu1110) [reply packages/components/socialFeed/NewsFeed/NewsFeed.tsx:38]: And my suggestion, the var name can be `isReady` =>
`const isReady = !(isFetching || isLoading)`

File: packages/screens/FeedPostView/FeedPostViewScreen.tsx [hunk 2]
```diff
@@ -60,17 +62,29 @@ export const FeedPostViewScreen: ScreenFC<"FeedPostView"> = ({
...
+    parentId: postResult?.identifier,
+    totalCount: postResult?.sub_post_length,
+  });
   const isNextPageAvailable = useSharedValue(hasNextPage);
 
+  const isLoadingPostValue = useMemo(
+    () => isLoadingPost || isFetchingPost,
+    [isLoadingPost, isFetchingPost]
+  );
+  const isLoadingCommentsValue = useMemo(
```
Comment (hthieu1110) [inline packages/screens/FeedPostView/FeedPostViewScreen.tsx:83]: not need useMemo

**Comment by sakul-budhathoki** [comment]
> 
Hi @omniwired  We are getting user info using useNSUserInfo, and those extra requests might be due to that. 
You can check the code below of SocialThreadCard.tsx
```
  const authorNSInfo = useNSUserInfo(localPost.createdBy);
  const userInfo = useNSUserInfo(wallet?.userId);
```

**Comment by omniwired** [comment]
@sakul-budhathoki is there anyway to limit / cache / reduce these kind of calls?

**Comment by n0izn0iz** [comment]
I'm sorry I didn't see this PR, I made the same kind of fix in #537, do we still need this PR, maybe it has other improvements? @WaDadidou

**Pull request closed without merging**